libs/src/services/discoveryProvider/index.js

```diff
@@ -676,11 +684,20 @@ class DiscoveryProvider {
       await this.autoSelectEndpoint()
     }
 
+    if (retries === 0) {
+      await this.autoSelectEndpoint(3, true)
```

Comment:
would be great if we could reset retries = initial value (3) here. That way we allow 3 fails on each discprov before trying to find another.